ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -537,12 +550,15 @@ module Vendor_nvidia = struct
     *)
     List.filter_map
       (fun pgpu ->
-        let devid = find_one_by_name "devId" pgpu in
-        if int_of_string (get_attr "deviceId" devid) = device_id then
-          let psubdev_id =
-            let id = int_of_string (get_attr "subsystemId" devid) in
-            if id = 0 then None else Some id
-          in
+        let devId = find_one_by_name "devId" pgpu in
+        let deviceId = int_of_string (get_attr "deviceId" devId) in
+        let subsystemId = int_of_string (get_attr "subsystemId" devId) in
+        let psubdev_id = if subsystemId = 0 then None else Some subsystemId in
```

Comment:
might be useful to add a comment here that `0` is a wildcard and thus ignored during matching.